packages/core/src/messages.ts

```diff
@@ -48,8 +60,14 @@ export function throwOnWarnings(hooks?: NestedHooks) {
     });
   } else {
     // Jest mode
-    beforeAll(() => hardFailMode++);
-    afterAll(() => hardFailMode--);
+    if ('beforeAll' in globalThis && 'afterAll' in globalThis) {
```

Comment:
We don't want this guard, it should crash if we get here with `hook` undefined and no `globalThis.beforeAll`.